### DISCUSSION TITLE
Possible collaboration/integration of the new OpenSilver 3.1?

### DISCUSSION THREAD

**Discussion by TheExiledCat** [AvaloniaUI/Avalonia#17761] [Ideas]
The people at opensilver: https://opensilver.net/announcements/3-1/
have finalized work on a vcscode plugin that allows you to now make XAML uis in vscode with drag and drop functionality.
This would ofcourse be really handy for avalonia in the future too and i wanted to make this post to see peoples interest and showcase this to the devs.

This would be a huge improvement not just on creation side of xaml but also a one up on the currect vscode plugin of avalonia and this doesnt just add previews but also the actual toolkit functionality

**Comment by stevemonaco** [comment]
I don't foresee any interest there as Avalonia is creating their own designer for Accelerate. I believe it's a standalone application, so it won't be tied to a specific IDE.

On the more opinionated side, VS Code tends to be a second-rate developer experience for C#. Most professional developers won't use it outside of web backend+frontend monorepos. Even then, many will split the web work to VS Code and backend to VS or Rider. This significantly diminishes the impact any system that's locked behind VS Code might have. I'm sure the Avalonia team has some internal metrics on usage across VS, Rider, and VS Code.
